### ISSUE TITLE
Pasting code results in crash

### ISSUE THREAD

**Issue by TeddyDD** [AlloyTools/org.alloytools.alloy#210]
```
Atom was 0
Exception: class java.lang.NullPointerException
Message: java.lang.NullPointerException: Failed to retrieve atom name.
Stacktrace:
class java.lang.NullPointerException: Failed to retrieve atom name.
java.desktop/sun.awt.X11.XlibWrapper.XGetAtomName(Native Method)
java.desktop/sun.awt.X11.XAtom.getName(XAtom.java:189)
java.desktop/sun.awt.X11.XDataTransferer.getTargetNameForAtom(XDataTransferer.java:172)
java.desktop/sun.awt.X11.XDataTransferer.getNativeForFormat(XDataTransferer.java:161)
java.desktop/sun.awt.datatransfer.DataTransferer.getFlavorsForFormats(DataTransferer.java:423)
java.desktop/sun.awt.datatransfer.ClipboardTransferable.<init>(ClipboardTransferable.java:88)
java.desktop/sun.awt.X11.XClipboard.getContents(XClipboard.java:108)
java.desktop/javax.swing.TransferHandler$TransferAction.actionPerformedImpl(TransferHandler.java:1755)
java.desktop/javax.swing.TransferHandler$TransferAction$1.run(TransferHandler.java:1712)
java.desktop/javax.swing.TransferHandler$TransferAction$1.run(TransferHandler.java:1710)
java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:97)
java.desktop/javax.swing.TransferHandler$TransferAction$2.run(TransferHandler.java:1730)
java.desktop/javax.swing.TransferHandler$TransferAction$2.run(TransferHandler.java:1728)
java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
java.desktop/javax.swing.TransferHandler$TransferAction.actionPerformed(TransferHandler.java:1727)
java.desktop/javax.swing.text.JTextComponent.invokeAction(JTextComponent.java:1523)
java.desktop/javax.swing.text.JTextComponent.paste(JTextComponent.java:1501)
edu.mit.csail.sdg.alloy4.OurSyntaxWidget$5.actionPerformed(OurSyntaxWidget.java:226)
java.desktop/javax.swing.SwingUtilities.notifyAction(SwingUtilities.java:1810)
java.desktop/javax.swing.JComponent.processKeyBinding(JComponent.java:2947)
java.desktop/javax.swing.JComponent.processKeyBindings(JComponent.java:2995)
java.desktop/javax.swing.JComponent.processKeyEvent(JComponent.java:2909)
java.desktop/java.awt.Component.processEvent(Component.java:6403)
java.desktop/java.awt.Container.processEvent(Container.java:2266)
java.desktop/java.awt.Component.dispatchEventImpl(Component.java:5001)
java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2324)
java.desktop/java.awt.Component.dispatchEvent(Component.java:4833)
java.desktop/java.awt.KeyboardFocusManager.redispatchEvent(KeyboardFocusManager.java:1952)
java.desktop/java.awt.DefaultKeyboardFocusManager.dispatchKeyEvent(DefaultKeyboardFocusManager.java:883)
java.desktop/java.awt.DefaultKeyboardFocusManager.preDispatchKeyEvent(DefaultKeyboardFocusManager.java:1150)
java.desktop/java.awt.DefaultKeyboardFocusManager.typeAheadAssertions(DefaultKeyboardFocusManager.java:1020)
java.desktop/java.awt.DefaultKeyboardFocusManager.dispatchEvent(DefaultKeyboardFocusManager.java:848)
java.desktop/java.awt.Component.dispatchEventImpl(Component.java:4882)
java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2324)
java.desktop/java.awt.Window.dispatchEventImpl(Window.java:2780)
java.desktop/java.awt.Component.dispatchEvent(Component.java:4833)
java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:773)
java.desktop/java.awt.EventQueue$4.run(EventQueue.java:722)
java.desktop/java.awt.EventQueue$4.run(EventQueue.java:716)
java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:97)
java.desktop/java.awt.EventQueue$5.run(EventQueue.java:746)
java.desktop/java.awt.EventQueue$5.run(EventQueue.java:744)
java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:743)
java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
```

```
Alloy Analyzer 6.1.0
Linux Void 6.1.21_1 x86_64 
$ java --version                                                                                                                                                                                                 3s
openjdk 17.0.5 2022-10-18
OpenJDK Runtime Environment (build 17.0.5+7-void-r2)
OpenJDK 64-Bit Server VM (build 17.0.5+7-void-r2, mixed mode, sharing)
```

**Comment by nmacedo** [comment]
Can you provide more information?

**Comment by TeddyDD** [comment]
Sure, what do you need?

**Comment by nmacedo** [comment]
Can tell us how to reproduce the error?

**Comment by TeddyDD** [comment]
Seems to be fixed in version 6.1.0 commit: 1bd6e4bfdf8f831e19ad6130cdb6cc4645eaee6e so closing for now.
